tests/test_api/test_bucketlist_items.py

```diff
@@ -0,0 +1,123 @@
+        self.assertIn("false", received_data)
+
+        # with a missing name /mispelled
+        payload = {'noname': 'cook breakfast',
+                   'nodescription': 'chapati stew'}
+        response = self.test_app.post(
+            '/api/v1/bucketlists/1/items/', data=payload, headers=self.header)
+        received_data = str(response.data, 'utf-8')
+        self.assertEqual(response.status_code, 400)
+        self.assertIn("name not provided", received_data)
+        # totally missing name
+        response = self.test_app.post(
+            '/api/v1/bucketlists/1/items/', headers=self.header)
+        received_data = str(response.data, 'utf-8')
+        self.assertEqual(response.status_code, 400)
+        self.assertIn("name not provided", received_data)
+
+        # creating an item in a bucketlist that doesn't exist
+        payload = {'name': 'swim',
+                   'description': 'go for swimming at ymca'}
+        response = self.test_app.post(
+            '/api/v1/bucketlists/13/items/', data=payload, headers=self.header)
+        received_data = str(response.data, 'utf-8')
+        self.assertEqual(response.status_code, 404)
+        self.assertIn("does not exist", received_data)
+
+    def test_bucket_editing_an_item(self):
+        # add an item
+        payload = {'name': 'cook lunch',
+                   'description': 'ugali omena for lunch'}
+        self.test_app.post(
+            '/api/v1/bucketlists/1/items/', data=payload, headers=self.header)
+
+        # edit item /name and done
+        payload = {'name': 'cooking lunch',
+                   'description': 'cooking ugali stew for lunch', 'done': True}
+        put_response = self.test_app.put(
+            '/api/v1/bucketlists/1/items/1', data=payload, headers=self.header)
+        received_data = str(put_response.data, 'utf-8')
+        self.assertEqual(put_response.status_code, 200)
+        self.assertIn("name", received_data)
+        self.assertIn("cooking lunch", received_data)
+        self.assertFalse("cook lunch" in received_data)
+        self.assertIn("done", received_data)
+        self.assertIn("true", received_data)
+        self.assertIn("cooking ugali stew for lunch", received_data)
+
+        # edit an item that does not exist
+        payload = {'name': 'cooking lunch', 'done': True}
+        put_response = self.test_app.put(
+            '/api/v1/bucketlists/1/items/13', data=payload, headers=self.header)
+        received_data = str(put_response.data, 'utf-8')
+        self.assertEqual(put_response.status_code, 404)
+        self.assertIn("does not exist", received_data)
+
+    def test_deleting_an_item_from_a_bucketlist(self):
+        # add an item
+        payload = {'name': 'cook lunch',
+                   'description': 'ugali omena for lunch'}
+        self.test_app.post(
+            '/api/v1/bucketlists/1/items/', data=payload, headers=self.header)
+        # delete item successfully
```

Comment:
minimise inline comments and prefer using docstrings.
 Code should be self documenting